e2e/tests-dfx/remote_generate_binding.bash

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bats
+
+load ../utils/_
+
+setup() {
+    standard_setup
+}
+
+teardown() {
+    standard_teardown
+}
+
+@test "generate-remote-binding succeeds on a simple remote example" {
+    install_asset remote_generate_binding/basic
+
+    assert_command dfx remote generate-binding --all
+
+    if [[ ! -f remote.mo ]]; then
+        echo "remote.mo not created"
+        exit 1
+    fi
+}
+
+@test "generate-remote-binding --overwrite behaves as expected" {
```

Comment:
So far the tests only check behavior with `--all` for a remote Motoko canister.  Let's fill out these tests for different combinations.  Here are a few ideas to get you started.  Please see what others can you think of!
- generate-binding <specific Motoko canister> (main file does not exist)
- generate-binding <specific Rust canister> (main file does not exist)
- generate-binding <specific canister> (main file already exists)
- generate-binding <specific canister> (multiple remote canisters exist; none have their main file already.  what happens?)
- generate-binding (no arguments; multiple remote canisters exist; none have their main file already.  what happens?)
- generate-binding --overwrite <specific canister> (main file already exists)
- generate-binding --all (multiple canisters: one already has its main file, another does not)

Reply:
Added tests for all those cases, plus what I can think of:
* `generate-binding --overwrite --all` behaves as expected